test/cypress/integration/manage-team-members/edit_permissions_spec.js

```diff
@@ -0,0 +1,123 @@
+'use strict'
+
+const SERVICE_EXTERNAL_ID = 'service_abc_123'
+const AUTHENTICATED_USER_ID = 'authenticated-user-id'
+const EDITING_USER_ID = 'user-we-are-editing-id'
+
+const getUserStubOpts = function (userExternalId, email, roleName, roleDescription) {
+  return {
+    external_id: userExternalId,
+    username: email,
+    email: email,
+    service_roles: [
+      {
+        service: {
+          external_id: SERVICE_EXTERNAL_ID
+        },
+        role: {
+          name: roleName,
+          description: roleDescription
+        }
+      }
+    ]
+  }
+}
+
+describe('Edit service user permissions', () => {
+  beforeEach(() => {
+    // keep the same session for entire describe block
+    Cypress.Cookies.preserveOnce('session')
+    Cypress.Cookies.preserveOnce('gateway_account')
+
+    const authenticatedUserStubOpts = getUserStubOpts(AUTHENTICATED_USER_ID, 'logged-in-user@example.com', 'admin', 'Administrator')
+    const userWeAreEditingStubOpts = getUserStubOpts(EDITING_USER_ID, 'other-user@example.com', 'admin', 'Administrator')
+    const viewOnlyUserStubOpts = getUserStubOpts('view-only-user-id', 'view-only-user@example.com', 'view-only', 'View Only')
+    const viewAndRefundUserStubOpts = getUserStubOpts('view-and-refund-user-id', 'view-and-refund-user@example.com', 'view-and-refund', 'View And Refund')
+
+    cy.task('setupStubs', [
+      {
+        name: 'getUserSuccess',
+        opts: authenticatedUserStubOpts
+      },
+      {
+        name: 'getUserSuccess',
+        opts: userWeAreEditingStubOpts
+      },
+      {
+        name: 'getServiceUsersSuccess',
+        opts: {
+          serviceExternalId: SERVICE_EXTERNAL_ID,
+          users: [
+            authenticatedUserStubOpts,
+            userWeAreEditingStubOpts,
+            viewOnlyUserStubOpts,
+            viewAndRefundUserStubOpts
+          ]
+        }
+      },
+      {
+        name: 'getInvitedUsersSuccess',
```

Comment:
why do we need this `getInvitedUsersSuccess` block? they seem redundant as the user we're editing is the one with email address other-user@example.com

Reply:
maybe not, I just thought I should do some basic tests on the pages we're viewing as they don't already exist as cypress tests. I could split these out into different spec files if you think this makes sense.
or if you don't think its worthwhile at all i could remove